src/ParallelAlgorithms/Amr/Criteria/Persson.hpp

```diff
@@ -0,0 +1,237 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <pup.h>
+#include <string>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/ValidateSelection.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Amr/Flag.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Options/Context.hpp"
+#include "Options/ParseError.hpp"
+#include "Options/String.hpp"
+#include "ParallelAlgorithms/Amr/Criteria/Criterion.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t>
+class ElementId;
+/// \endcond
+
+namespace amr::Criteria {
+
+/// @{
+/*!
+ * \brief Computes an anisotropic smoothness indicator based on the power in the
+ * highest modes
+ *
+ * This smoothness indicator is the L2 norm of the tensor component with the
+ * lowest N - M modes filtered out, where N is the number of grid points in the
+ * given dimension and M is `num_highest_modes`.
+ *
+ * This strategy is similar to the Persson troubled-cell indicator (see
```

Comment:
Can you add some math here? This is similar to Persson in the sense of it uses spectral coefficients, but you have completely different parameters from Persson, so you really need to document this. I don't understand what value I should choose when compare to the alpha value of the Persson TCI. That needs to be made extremely clear because that value is central to the Persson TCI.

Reply:
relative tolerance = N^{-\alpha}

I'll just switch to `Alpha` as a parameter.